store.py

```diff
@@ -22,65 +22,182 @@
 from __future__ import annotations
 from typing import List, Optional, TextIO
 import json
-
 # Use this constant in your code
 EXPRESS_LIMIT = 7
 
 
-# TODO: Complete the GroceryStore class and methods according to the docstrings
-# You may add private attributes and helper methods, but do not change the
-# public interface.
-# Write docstrings for all methods you write, and document your attributes
-# in the class docstring.
 class GroceryStore:
     """A grocery store.
+
+    === Attributes ===
+    _regular_count: How many regular lines are open.
+    _express_count: How many express lines are open.
+    _self_serve_count: How many self-serve lines are open.
+    _line_capacity: maximum amount of people allowed in each line
+    _line_list: list of all the lines open, following representation invariants
+
+    === Representation Invariant ===
+    - _line_list is ordered in the following order:
+    RegularLine, ExpressLine, SelfServeLine
+    - _line_capacity is the same for all lines
     """
+    _regular_count: int
+    _express_count: int
+    _self_serve_count: int
+    _line_capacity: int
+    _line_list: List[Any]
 
     def __init__(self, config_file: TextIO) -> None:
         """Initialize a GroceryStore from a configuration file <config_file>.
+        >>> import io
+        >>> config_file = io.StringIO('{"regular_count":0,"express_count":0,"self_serve_count":1,"line_capacity":10}')
+        >>> g = GroceryStore(config_file)
+        >>> g.get_info('_regular_count') == 0
+        True
+        >>> g.get_info('_express_count') == 0
+        True
+        >>> g.get_info('_self_serve_count') == 1
+        True
+        >>> g.get_info('_line_capacity') == 10
+        True
+        """
+        working = json.load(config_file)
+        self._regular_count = working.get('regular_count')
+        self._express_count = working.get('express_count')
+        self._self_serve_count = working.get('self_serve_count')
+        self._line_capacity = working.get('line_capacity')
+
+        self._line_list = []
+        i = 0
+        while i < self._regular_count:
```

Comment:
Why are using "while" loop here ? using "for" loop would be much easier condition-wise and you do not have to track conditions in a crazy way as the in 2 cycles below